scripts/imgtool/main.py

```diff
@@ -154,6 +154,23 @@ def validate_version(ctx, param, value):
         raise click.BadParameter("{}".format(e))
 
 
+def validate_security_counter(ctx, param, value):
```

Comment:
I wonder if we should use something like `-s auto` instead of just always auto generating the security counter. If the feature is not enabled, this just wastes space in the TLV.

Reply:
> If the feature is not enabled, this just wastes space in the TLV.

If you don't pass `-s` no TLV is added.

Reply:
Yes. To auto generate the security counter from the image version an empty string had to be passed, but I changed it to `auto` (really a better solution).